monkey/tests/unit_tests/infection_monkey/network/relay/test_relay_connection_handler.py

```diff
@@ -0,0 +1,64 @@
+import socket
+from ipaddress import IPv4Address
+from unittest.mock import MagicMock
+
+import pytest
+
+from monkey.infection_monkey.network.relay import (
+    RelayConnectionHandler,
+    RelayUserHandler,
+    TCPPipeSpawner,
+)
+from monkey.infection_monkey.network.relay.relay_connection_handler import RELAY_CONTROL_MESSAGE
+
+USER_ADDRESS = "0.0.0.1"
+
+
+@pytest.fixture
+def pipe_spawner():
+    return MagicMock(spec=TCPPipeSpawner)
+
+
+@pytest.fixture
+def relay_user_handler():
+    return MagicMock(spec=RelayUserHandler)
+
+
+@pytest.fixture
+def close_socket():
+    sock = MagicMock(spec=socket.socket)
+    sock.recv.return_value = RELAY_CONTROL_MESSAGE
+    sock.getpeername.return_value = (USER_ADDRESS, 12345)
+    return sock
+
+
+@pytest.fixture
+def data_socket():
+    sock = MagicMock(spec=socket.socket)
+    sock.recv.return_value = b"some data"
+    sock.getpeername.return_value = (USER_ADDRESS, 12345)
+    return sock
+
+
+def test_control_message_disconnects_user(pipe_spawner, relay_user_handler, close_socket):
+    connection_handler = RelayConnectionHandler(pipe_spawner, relay_user_handler)
+
+    connection_handler.handle_new_connection(close_socket)
+
+    relay_user_handler.disconnect_user.assert_called_once_with(IPv4Address(USER_ADDRESS))
+
+
+def test_connection_spawns_pipe(pipe_spawner, relay_user_handler, data_socket):
+    connection_handler = RelayConnectionHandler(pipe_spawner, relay_user_handler)
+
+    connection_handler.handle_new_connection(data_socket)
+
+    assert pipe_spawner.spawn_pipe.called
```

Comment:
Should we `assert_called_once_with(data_socket)`?

Reply:
I was on the fence about using `assert_called_once_with()` for any of the tests as I was concerned about leaking too much about the internals in the tests. However, I think we're fine using it. After all, how else would it use `PipeSpawner`?